src/index.ts

```diff
@@ -1,93 +1,215 @@
+  }
 
-  _areas = [];
-  _lecturers = []; // Name, surname, position, company, experience, courses, contacts
+  get lecturers(): Lecturer[] {
+    return this._lecturers;
+  }
+
+  addArea(areaToAdd: Area): void {
+    this._areas.push(areaToAdd)
+  }
 
-  get areas() {
-    this._areas;
+  removeArea(areaToRemove: Area): void {
+    this._areas = this._areas.filter((area: Area): boolean => area !== areaToRemove)
   }
 
-  get lecturers() {
-    this._lecturers;
+  addLecturer(lecturerToAdd: Lecturer): void {
+    this._lecturers.push(lecturerToAdd)
+  }
+
+  removeLecturer(lecturerToRemove: Lecturer): void {
+    this._lecturers = this._lecturers.filter((lecturer: Lecturer): boolean => lecturer !== lecturerToRemove)
   }
 }
 
 class Area {
   // implement getters for fields and 'add/remove level' methods
-  _levels = [];
-  _name;
+  _levels: Level[] = [];
+  _name: string;
 
-  constructor(name) {
+  constructor(name: string) {
     this._name = name;
   }
+
+  get levels(): Level[] {
+    return this._levels
+  }
+
+  get name(): string {
+    return this._name
+  }
+
+  addLevel(levelToAdd: Level): void {
+    this._levels.push(levelToAdd)
+  }
+
+  removeLevel(levelToRemove: Level): void {
+    this._levels = this._levels.filter((level: Level): boolean => level !== levelToRemove)
+  }
 }
 
 class Level {
   // implement getters for fields and 'add/remove group' methods
 
-  _groups;
-  _name;
+  _groups: Group[] = [];
+  _name: string;
+  _description: string;
 
-  constructor(name, description) {
-    this.name = name;
+  constructor(name: string, description: string) {
+    this._name = name;
     this._description = description;
   }
+
+  get name(): string {
+    return this._name
+  }
+
+  get description(): string {
+    return this._description
+  }
+
+  get groups(): Group[] {
+    return this._groups
+  }
+
+  addGroup(groupToAdd: Group): void {
+    this._groups.push(groupToAdd)
+  }
+
+  removeGroup(groupToRemove: Group): void {
+    this._groups = this._groups.filter((group: Group): boolean => group !== groupToRemove)
+  }
 }
 
 class Group {
   // implement getters for fields and 'add/remove student' and 'set status' methods
 
-  _area;
-  _status;
-  _students = []; // Modify the array so that it has a valid toSorted method*
+  _area: Area | undefined;
+  _status: string | undefined;
+  _students: Student[] = []; // Modify the array so that it has a valid toSorted method*
+  directionName: string;
+  levelName: typeof Level.name;
 
-  constructor(directionName, levelName) {
+  constructor(directionName: string, levelName: typeof Level.name) {
     this.directionName = directionName;
     this.levelName = levelName;
   }
 
-  showPerformance() {
-    const sortedStudents = this._students.toSorted((a, b) => b.getPerformanceRating() - a.getPerformanceRating());
+  get area(): Area | "There is no area selected"  {
+    if (this._area) {
+      return this._area
+    } else {
+      return "There is no area selected"
```

Comment:
It's better to define enums or constants for error messages. This makes the code more efficient and reusable.